src/PersistentFilters.ts

```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+            return fileHistoryItem.includes(name.toUpperCase());
+        });
+        if (index >= 0) { this.mFileHistory.splice(index, 1); }
+        await this.updateFileHistory();
     }
 
+    /*********************************************************************************************************************************************/
+    /* Reset functions, for resetting the persistent array to empty (in the extension and in settings.json)
+    /*********************************************************************************************************************************************/
+
+    /**
+     * Empties the persistent search history array
+     * Sets the extension's searchHistory array to empty, and updates the array in settings.json
+     *
+     */
+    public async resetSearchHistory() {
+        this.mSearchHistory = [];
+        this.updateSearchHistory();
+    }
+
+    /**
+     * Empties the persistent sessions array
+     * Sets the extension's sessions array to empty, and updates the array in settings.json
+     *
+     */
     public async resetSessions() {
         this.mSessions = [];
         this.updateSessions();
     }
 
     /**
-     * Initializes the history and sessions sections by reading from a file
+     * Empties the persistent recently-opened files history array
+     * Sets the extension's fileHistory array to empty, and updates the array in settings.json
+     *
      */
-    private async initialize() {
-        let lines: string[];
-        let recallLines: string[];
-        if (vscode.workspace.getConfiguration(this.schema)) {
-            lines = vscode.workspace.getConfiguration(this.schema).get(PersistentFilters.history);
-            recallLines = vscode.workspace.getConfiguration(this.schema).get(PersistentFilters.recall);
-        }
-        if (recallLines) {
-            this.mRecall = recallLines;
-        }
-        if (lines) {
-            this.mHistory = lines;
-        } else {
-            this.resetHistory();
-        }
-        if (vscode.workspace.getConfiguration(this.schema)) {
-            lines = vscode.workspace.getConfiguration(this.schema).get(PersistentFilters.sessions);
-        }
-        if (lines) {
-            this.mSessions = lines;
-        } else {
-            this.resetSessions();
-        }
+    public async resetFileHistory() {
+        this.mFileHistory = [];
+        this.updateFileHistory();
     }
 
-    private async updateHistory() {
-        // settings are read-only, so make a clone
+    /*********************************************************************************************************************************************/
+    /* Update functions, for updating the settings.json file in VSCode
+    /*********************************************************************************************************************************************/
+
+    /**
+     * Updates the stored favorites array in settings.json
+     */
```

Comment:
I think the block comment above + function name makes it clear and this comment is redundant.

Reply:
Looking at this again, alternatively removing the block comment is a better solution since this jsdoc shows as a hint in intellisense.

Reply:
I would like to leave the block comment, as it helps organize the file. Each set of functions in the file under a block comment have similar functions. So when you are scrolling around you can quickly get to the section you are looking for.

The Intellisense thing is also useful.